### ISSUE TITLE
Switch among open windows with Super+1..9 keyboard shortcut

### ISSUE THREAD

**Issue by dandv** [mate-desktop/marco#283]
TL;DR - This is a productivity feature that saves me about an hour a day, and is implemented by Unity, Windows and in progress in KDE and Cinnamon.

---

The Super/Meta/Win+1..9 keyboard shortcuts are especially handy if you often switch among the same few apps, e.g. a browser, a file manager and an editor/IDE, but also have other apps open, so Alt+Tabbing to the desired window would take more time.[1]

That keyboard shortcut saves me probably 5 seconds per minute on the average, so ~40 minutes a typical 8-hour day, and it's supported by Unity and Windows 7 as well.
## Workarounds

Adding custom keyboard shortcuts for individual apps doesn't work, because pressing the shortcut will launch a new instance of the app.[2]

[1]: DEs will order the most frequently used windows first, but you have to first look and see how many times you have to press Alt+Tab; while with Super+1 or Super+2, you always know which shortcut goes to one app or the other.

[2]: Some apps have an option to allow only one instance, but this is the exception rather than the rule, and even in that case, there will be a noticeable flicker of launching the new instance and seeing it exiting immediately.
## Demand for this feature
- Windows supports it
- Unity supports it
- Cinnamon users want it - https://github.com/linuxmint/Cinnamon/issues/5561
- [KDE is working on it](https://bugs.kde.org/show_bug.cgi?id=336348)
- http://askubuntu.com/questions/813262/which-desktop-environments-allow-switching-among-windows-with-supernumbers
- I've placed a $100 bounty on it
## 

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/37703001-switch-among-open-windows-with-super-1-9-keyboard-shortcut)**

We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F841233&utm_medium=issues&utm_source=github).

**Comment by vkareh** [comment]
@dandv - [this fix](https://github.com/robint99/mate-dock-applet/issues/91) for [mate-dock-applet](https://github.com/robint99/mate-dock-applet/) gives you Super+1/2/3/etc keybindings for open apps.

You can also pin them to specific positions (I find that most useful for quickly launching my everyday apps in the morning).

Pressing the combination for an already open app will let you cycle through all windows of said app.

**Comment by vkareh** [comment]
Since this is feature is provided by a popular extension, I'm closing this issue.
Let me know if you disagree with this or want to discuss other options!
Thanks!

**Comment by vkareh** [comment]
That's a fair point, so I gave this some thought.

The technical aspects of mapping windows/apps to Super+# are well understood, so that's not a problem.

I think that the problem comes from deciding which shortcut maps to which window. Unity only has the one launcher/dock on the side, so the mapping is in order (top to bottom) and you can pin applications that will open with the same key combo.

MATE, however, is much more configurable than that. You can have multiple panels, each with their own application launchers, either as individual icons or as dock applets. Then there's also the traditional window list (of which you can have multiple, each on a different screen listing different applications).

So, which of all those is the one receiving the Super+# event? My guess would be the window list on the bottom panel, because that's the layout I use, but other users might disagree. But say we choose that one, now we can change between open applications, but we cannot open them from closed. We still have the problem of multiple window lists on multiple panels/monitors.

If we choose marco, then we could use the window list from Alt+Tab. However, that still means we cannot open applications, just switch on already-open ones, but now we also don't have a visual cue as to which window we will be selecting, so there will be a lot of trial and error from the user every time a window is closed. On top of that, the marco window switcher reorders the windows based on last focus. What this means is that which number to use for which window will actually change every time you change focus, which would make it effectively as good as randomly selecting a window from the list and hoping it's the one you want.

So what do we need then? Well:
* we need a visual cue of which windows we can select
* we need to be able to open applications from closed
* we need the position of that application to be static with respect to the others

If we forego the ability to open applications, we could use the traditional window list, and only select from the currently-active monitor if there are more than one window list applets.

Everything else seems like the kind of features that a dock would be able to provide. We have several of those: plank, docky, mate-dock-applet, etc... This last one very closely emulates what Unity does with their launcher (so much so that Ubuntu MATE includes it by default and is part of their Unity-replacement layout, Mutiny)

I'm more than happy to consider alternatives that would improve on other people's workflows. I'm currently very satisfied with mate-dock-applet and would consider the other options posted above a downgraded experience, but there are probably other ways that mate/marco could do it that I haven't thought about :/

**Comment by guglez** [comment]
Hi All,


I have Linux Mint 18.3 and MATE Dock Applet v0.80. Keyboard shortcuts (super+number key) are working but after some time they just stop to work. I have to restart my session to make them work again. Is there any way to fix that?

**Comment by vkareh** [comment]
@guglez - this was fixed on v0.86 of the mate-dock-applet. Here: https://github.com/robint99/mate-dock-applet/commit/7a9fa4ea6e757c14a02285eb8f0b1a844a3e84d5

**Comment by mattarroz** [comment]
> If we forego the ability to open applications, we could use the traditional window list, and only select from the currently-active monitor if there are more than one window list applets.

@vkareh  Honestly, for me that would be completely fine. I tried various docks but wasn't really happy with them. For me basically the only thing that's missing from the traditional list is the shortcuts. If you give me some hints, I would be happy to implement this and create a PR.
